promtimer/promtimer.py

```diff
@@ -252,7 +287,58 @@ def main():
     grafana_port = args.grafana_port
     prometheus_base_port = grafana_port + 1
 
-    if not args.cluster:
+    cbbackupmgr_stats_mode = False
+    buckets = None
+
+    if args.stats_archive_path:
+        cbbackupmgr_stats_mode = True
+
+        archive_path = args.stats_archive_path
+        if zipfile.is_zipfile(args.stats_archive_path):
+            archive_path_tmpdir = tempfile.TemporaryDirectory()
+            atexit.register(lambda: archive_path_tmpdir.cleanup())
+            archive_path = archive_path_tmpdir.name
+
+            with zipfile.ZipFile(args.stats_archive_path, 'r') as zipped_archive:
+                zipped_archive.extractall(archive_path)
+
+            archive_path = os.path.join(archive_path, os.path.splitext(os.path.basename(args.stats_archive_path))[0])
+        # Check that stats_archive_path exists and is not an empty dir
+        elif not os.path.isdir(args.stats_archive_path) or \
+            len(os.listdir(args.stats_archive_path)) == 0:
+            logging.error(
+                'directory supplied as stats_archive_path either does not exist or is empty'
+            )
+            sys.exit(1)
+
+        stats_tsdb_path_tmpdir = tempfile.TemporaryDirectory()
+        atexit.register(lambda: stats_tsdb_path_tmpdir.cleanup())
```

Comment:
I'd not bother with this. I think it's fine to leave things in `/tmp` hanging around

Reply:
I'd vote the other way - obviously can't guarantee it'll be cleaned up (e.g., if we're SIGKILL-ed) but seems nice to make an effort to do so on a routine exit.

Reply:
Have removed the lambda